projects/admin/src/app/routes/import-documents-route.ts

```diff
@@ -45,58 +62,51 @@ export class ImportDocumentsRoute extends BaseRoute implements RouteInterface {
    * @return Object
    */
   getConfiguration() {
-    return {
+    const config = {
       path: 'records/:type',
       children: [
         { path: '', component: RecordSearchPageComponent },
         { path: 'detail/:pid', component: DetailComponent }
       ],
       data: {
-        types: [
-          {
-            key: 'import_bnf',
-            label: 'BNF Importation',
-            component: DocumentsBriefViewComponent,
-            detailComponent: DocumentDetailViewComponent,
-            canAdd: () => of(false),
-            canUpdate: () => of(false),
-            canDelete: () => of(false),
-            resultsText: (hits: any) => this.getResultsText(hits),
-            aggregationsBucketSize: 10,
-            aggregationsOrder: [
-              'document_type',
-              'author',
-              'language',
-              'year'
-            ],
-            aggregationsExpand: ['document_type', 'author', 'language', 'year'],
-            itemHeaders: {
-              Accept: 'application/rero+json, application/json'
-            },
-            listHeaders: {
-              Accept: 'application/rero+json, application/json'
-            },
-            allowEmptySearch: false
-          }
-        ]
+        types: []
       }
     };
+    this._routeToolService.injector.get(ImportSourceApiService)
+      .getSources()
+      .subscribe((sources: Array<ExternalSourceSetting>) => {
```

Comment:
I would use  a pipe, map.

Reply:
The current solution is the best one according to the time available.